### DISCUSSION TITLE
Reporter with JS/TS Source Context on Failure?

### DISCUSSION THREAD

**Discussion by mlynch** [webdriverio/webdriverio#7926] [Q&A]
Hi,

I'm looking for a reporter that can show test source code context when a test fails. For example, Cypress has this feature:

<img width="1451" alt="image" src="https://user-images.githubusercontent.com/11214/149666410-1d5d029a-9bd7-4d5b-9e97-2e79c202345c.png">

Is this possible in wdio today or in any one of the reporters? Or, do I need to somehow enable sourcemaps in my app? I've been following the wdio TypeScript process which works well minus source context.

**Comment by christian-bromann** [comment]
Hey @mlynch,

WebdriverIO has the error stack available to expose such data but afaik non of the reporters do atm. Would be a nice addition to see the code lines that fail.
